### PR TITLE
chore(docs): document how to correctly override list(string) parameters

### PR DESCRIPTION
**Pull request by johnstcn**
While investigating https://github.com/coder/coder/issues/15488 I found it _really_ tricky to figure out how to properly override a `list(string)`.

Documenting my findings for great justice.

### CODE REVIEW

File: cli/create_test.go [hunk 1]
```diff
@@ -864,24 +868,55 @@ func TestCreateValidateRichParameters(t *testing.T) {
...
+		t.Run("Default", func(t *testing.T) {
+			t.Parallel()
+			inv, root := clitest.New(t, "create", "my-workspace-2", "--template", template.Name, "--yes")
+			clitest.SetupConfig(t, member, root)
+			clitest.Run(t, inv)
+		})
+
+		t.Run("CLIOverride/DoubleQuote", func(t *testing.T) {
+			t.Parallel()
+			inv, root := clitest.New(t, "create", "my-workspace-3", "--template", template.Name, "--parameter", "\""+listOfStringsParameterName+"=[\"\"ddd=foo\"\",\"\"eee=bar\"\",\"\"fff=baz\"\"]\"", "--yes")
```
Comment (mtojek) [inline cli/create_test.go:901]: Can you use backticks instead so this is human readable?

File: docs/admin/templates/extending-templates/parameters.md [hunk 2]
```diff
@@ -79,6 +79,34 @@ data "coder_parameter" "security_groups" {
...
+> you will need to take care of both CSV quoting and shell quoting.
+>
+> For the above example, to override the default values of the `security_groups`
+> parameter, you will need to pass the following argument to `coder create`:
+>
+> ```
+> --parameter "\"security_groups=[\"\"DevOps Security Group\"\",\"\"Backend Security Group\"\"]\""
+> ```
+>
+> You can use [this Go Playground link](https://go.dev/play/p/yvI9rdtS0ch) to
```
Comment (mtojek) [inline docs/admin/templates/extending-templates/parameters.md:97]: Security question: is it safe to leave it as is? Do these links expire? Maybe just drop a snippet around docs.

File: cli/create_test.go [hunk 3]
```diff
@@ -864,24 +868,55 @@ func TestCreateValidateRichParameters(t *testing.T) {
...
-			if value != "" {
-				pty.WriteLine(value)
+			matches := []string{
+				listOfStringsParameterName, "",
+				"aaa, bbb, ccc", "",
+				"Confirm create?", "yes",
 			}
-		}
+			for i := 0; i < len(matches); i += 2 {
+				match := matches[i]
```
Comment (mtojek) [inline cli/create_test.go:883]: In [this PR](https://github.com/coder/coder/pull/15406) Spike replaced the for-loop with a ladder of matches. IMHO it is a more reader-friendly form.

**Review by mtojek — APPROVED** [review]
👍

File: docs/admin/templates/extending-templates/parameters.md (hunk 2 above)
Comment (johnstcn) [reply docs/admin/templates/extending-templates/parameters.md:97]: Although I do see it's not very 'user-friendly' to have to edit some Go code to figure something out. I'll move the playground link to the corresponding test instead but keep the above example invocation generated by the playground link.

File: cli/create_test.go [hunk 4]
```diff
@@ -864,24 +868,55 @@ func TestCreateValidateRichParameters(t *testing.T) {
...
+				b  bytes.Buffer
+				sb strings.Builder
+			)
+			require.NoError(t, json.NewEncoder(&b).Encode([]string{"ddd=foo", "eee=bar", "fff=baz"}))
+			cw := csv.NewWriter(&sb)
+			require.NoError(t, cw.Write([]string{listOfStringsParameterName + "=" + b.String()}))
+			cw.Flush()
+			require.NoError(t, cw.Error())
+			t.Logf("it looks like this: \n%q", strings.TrimSpace(sb.String()))
+		})
```
Comment (johnstcn) [inline cli/create_test.go:919]: this is actually redundant in favour of the playground link, so will remove

File: cli/create_test.go (hunk 1 above)
Comment (johnstcn) [reply cli/create_test.go]: Done!

File: cli/create_test.go (hunk 3 above)
Comment (johnstcn) [reply cli/create_test.go]: Done! 🪜

**Pull request merged**